### PR TITLE
Local storage for settings and theme

### PR DESCRIPTION
**Pull request by fairnightzz**

### CODE REVIEW

File: src/components/MangaControl.tsx [hunk 1]
```diff
@@ -35,9 +35,10 @@ type Props = {
 }
 
 const MangaControl: React.FunctionComponent<Props> = (props: Props) => {
-  const [menu, setMenu] = useState(true)
   const { readerSettings, updateReaderSettings, chapter, chapters, updateChapter, page, pages, updatePage } = props;
-
+  const changeMenuState = () => {
```
Comment (MrPicklePinosaur) [inline src/components/MangaControl.tsx]: maybe rename to `toggleMenuState`

File: src/pages/Reader.tsx [hunk 2]
```diff
@@ -30,6 +30,35 @@ class Reader extends React.Component<any, State> {
...
   constructor(props: any) {
     super(props);
+    let readerSettings = {
+      readerMode: ReaderMode.longStrip,
+      imageSizing: ImageSizing.fitHeight,
+      colourTheme: ColourTheme.light,
+      menu: Menu.open,
+    }
+
+    const savedSettings = localStorage.getItem('carrotSettings');
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:40]: possibly overkill but can also define constants for each localStorage key

File: src/pages/Reader.tsx [hunk 3]
```diff
@@ -30,6 +30,35 @@ class Reader extends React.Component<any, State> {
...
+    if (savedSettings) {
+      try {
+        const parsedSettings = JSON.parse(savedSettings)
+        if (!parsedSettings.readerMode) {
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx]: isnt all these redundant? you already set all the default values of the settings when declaring `readerSettings`

i would rephrase the whole thing as something like:
```
if (parsedSettings?.readerMode !== undefined) {//or whatever null checking u want
  readerSettings.readerMode == parsedSettings.readerMode
}
```
alternatively, you could also store each setting as a seperate localstorage item, although that would probably get too messy

File: src/pages/Reader.tsx (hunk 2 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx:40]: Might be a consideration for the future. I suggest keeping everything together though, because otherwise we would have to pass every single individual setting as a prop - or the reader settings just holds the functions so its more clean

File: src/pages/Reader.tsx (hunk 3 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx]: no because an evil person would malform the contents of `carrotSettings`, resulting in a failed json parse which is why i try catch it

File: src/pages/Reader.tsx [hunk 4]
```diff
@@ -135,6 +160,7 @@ class Reader extends React.Component<any, State> {
       ...curState,
       readerSettings: newReaderSettings
     }));
+    localStorage.setItem("carrotSettings", JSON.stringify(newReaderSettings))
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:162]: not necessarily needed but possibly check that newReaderSettings doesn't have any weird values before dumping it into localstorage?

**Review by MrPicklePinosaur — CHANGES_REQUESTED** [review]
clean up the reading from carrotSettings code

File: src/pages/Reader.tsx (hunk 4 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx:162]: not too particularly worried about dumping into localstorage, more worried about getting because its hard to modify react state in production

File: src/pages/Reader.tsx (hunk 2 above)
Comment (MrPicklePinosaur) [reply src/pages/Reader.tsx:40]: just meant do sm like
```
export const carrotSettingsKey = 'carrotSettings';
```
and then
```
localStorage.getItem(carrotSettingsKey)
```

File: src/pages/Reader.tsx (hunk 3 above)
Comment (MrPicklePinosaur) [reply src/pages/Reader.tsx]: no you can keep the try catch, just the
```
if (!parsedSettings.readerMode)
```
stuff is redundant

File: src/pages/Reader.tsx (hunk 4 above)
Comment (MrPicklePinosaur) [reply src/pages/Reader.tsx:162]: ok not too important ig

File: src/pages/Reader.tsx (hunk 2 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx:40]: meh
